pkg/cmd/ssh/options.go

```diff
@@ -491,6 +491,10 @@ func (o *SSHOptions) Run(f util.Factory) error {
 		return nil // abort
 	}
 
+	if !shoot.Spec.Provider.WorkersSettings.SSHAccess.Enabled {
```

Comment:
The properties `WorkersSettings` and `SSHAccess` are pointers. They are optional and can be nil! Should take care fore these cases. It would also be nice to have a unit test for this.
```golang
    workersSettings := shoot.Spec.Provider.WorkersSettings
    if workersSettings != nil && workersSettings.SSHAccess != nil && workersSettings.SSHAccess.Enabled {
		return errors.New("Node SSH access disabled, SSH not allowed")
	}
```

Reply:
done 975216a @holgerkoser